app/api/lectures.js

```diff
@@ -0,0 +1,276 @@
+    // extract ids of returned sections
+    let section_ids = []
+    for (let i = 0; i < found_sections.length; i++) {
+        section_ids.push(found_sections[i].id)     
+    }
+    return section_ids
+}
+
+async function getSectionLectureRelation (lectureId, sectiondId) {
+    return await db.LectureForSection.findOne({
+        where: {
+            lectureId: lectureId,
+            sectionId: sectiondId
+        }
+    })
+}
+
+// get all lecture objects for the current course
+router.get('/', requireAuthentication, async function (req, res) {
+    const user = await db.User.findByPk(req.payload.sub)
+    const courseId = req.params.course_id
+    const enrollment = await getEnrollmentFromCourse(user.id, courseId) || await getEnrollmentFromSectionInCourse(user.id, courseId)
+    const course = await getCourse(courseId)
+
+    if (enrollment == null) {   // if user is not enrolled in this course
+        res.status(403).send()
+    }
+    else if (enrollment.role == 'student' && !course.published) {   // if user is a student, and course isn't published, return 'No Content'
+        res.status(204).send()
+    }
+    else {  // if teacher, OR student in published course
+        const lectures = await db.Lecture.findAll({
+            where: { courseId: courseId }
+        })
+        if (lectures == []) {   // if no lectures are in this course
+            res.status(204).send()
+        }
+        else {
+            res.status(200).json({
+                "lecture": lectures
+            })
+        }
+    }
+})
+
+router.post('/', requireAuthentication, async function (req, res) {
+    const user = await db.User.findByPk(req.payload.sub)
+    const courseId = req.params.course_id
+    const enrollment = await getEnrollmentFromCourse(user.id, courseId) || await getEnrollmentFromSectionInCourse(user.id, courseId)
+    var lecture     // will hold the returned lecture object from database
+
+    if (enrollment == null) {   // if user is not enrolled in this course
+        res.status(403).send()
+    }
+    else if (enrollment.role == 'teacher') {
+        // create lecture object
+        let newLec = req.body
+        newLec['courseId'] = courseId
+        try {
+            lecture = await db.Lecture.create(newLec)
+        }
+        catch (e) {
+            res.status(400).send({error: "Unable to create lecture object"})
+            return
+        }
+
+        // create lecture-section association for all sections in course
+        const sectionIds = await getSectionsIdsFromCourse(courseId);
+        try {
+            // iterate through each section in this course and add relationship
+            for (let i = 0; i < sectionIds.length; i++) {
+                await db.LectureForSection.create({
+                    lectureId: lecture.id,
+                    sectionId: sectionIds[i]
+                    // TODO: no other fields are being added here
+                    // ...should it be added here, or in /:section_id/lectures ?
```

Comment:
I think this is right. We would want to add this whenever a lecture is added to the course or whenever a section is added to the course